webapp/src/components/modals/create_issue/create_issue.jsx

```diff
@@ -105,7 +105,18 @@ export default class CreateIssueModal extends PureComponent {
         this.setState(initialState, this.props.close);
     };
 
-    handleRepoChange = (repo) => this.setState({repo});
+    handleRepoChange = (repo) => {
+        const fieldsToReset = {
+            repo: repo,
+            issueTitle: '',
+            issueDescription: '',
+            labels: [],
+            assignees: [],
+            milestone: null,
+            showErrors: false,
+        }
+        this.setState(fieldsToReset);
```

Comment:
I updated this since I noticed that on switching repos previously selected labels didn't reset.

Reply:
Or maybe this

```suggestion
        this.setState({
            ...initialState,
            repo,
            issueTitle: this.state.issueTitle,
            issueDescription: this.state.issueDescription,
        });
```

Reply:
I would go with the second one. Commiting it.